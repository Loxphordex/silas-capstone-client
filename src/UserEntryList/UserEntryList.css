#user-entry-list li {
  list-style: none;
}