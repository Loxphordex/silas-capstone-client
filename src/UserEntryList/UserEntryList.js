import React from 'react'
import ApiService from '../services/api-service'
import TokenService from '../services/token-service'
import './UserEntryList.css'

export default class UserEntryList extends React.Component {

  componentDidMount() {
    if(TokenService.getAuthToken()) {
      ApiService.getUserEntries()
        .then(entries => {
          this.props.updateUserEntries(entries)
        })
    }
  }

  handleDelete = (id) => {
    if(TokenService.getAuthToken()) {
      ApiService.deleteEntry(id)
        .then(() => {
          ApiService.getUserEntries()
            .then(entries => {
              this.props.updateUserEntries(entries)
            })
        })
    }
  }

  render() {
    return (
      <ul id='user-entry-list'>
        {this.props.userEntries && this.props.userEntries.length > 0 && this.props.userEntries.map(entry => {
          return (
            <li key={entry.id}>
              <div id={entry.id} className='entry-list-item' onClick={() => this.props.handleGetEntry(entry.id)}>
                <h3 id={entry.title}>{entry.title}</h3> 
              </div>
              <button className='delete-button' onClick={() => this.handleDelete(entry.id)}>
                <i className="fa fa-trash"></i>
              </button>
            </li>
          )
        })}
      </ul>
    )
  }
}