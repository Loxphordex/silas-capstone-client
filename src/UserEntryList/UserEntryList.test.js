import React from 'react';
import UserEntryList from './UserEntryList'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Login Page', () => {
    it('renders the complete page', () => {
    const wrapper = shallow(<UserEntryList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})