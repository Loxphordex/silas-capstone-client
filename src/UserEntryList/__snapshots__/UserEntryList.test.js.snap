// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page renders the complete page 1`] = `
<ul
  id="user-entry-list"
/>
`;
